### PR TITLE
Improving exception msgs for database types

### PR DESCRIPTION
**Pull request by burzum**
It's annoying that you can't see the type and make a guess which of the many values it might be.

### CODE REVIEW

File: src/Database/Type/BoolType.php [hunk 1]
```diff
@@ -67,7 +67,10 @@ public function toDatabase($value, Driver $driver)
             return (bool)$value;
         }
 
-        throw new InvalidArgumentException('Cannot convert value to bool');
+        throw new InvalidArgumentException(sprintf(
+            'Cannot convert value of type `%s` to bool',
+            getTypeName($value)
```
Comment (dereuromark) [inline src/Database/Type/BoolType.php]: Wouldnt it then also make sense to include the value itself directly? Or is that a bit too much then as debug info?

Comment (burzum) [reply src/Database/Type/BoolType.php]: I wanted to do that, but how could we archive this? It's just the base \InvalidArgumentException and doesn't really have a way to attach arbitrary data to it? Also does the error renderer has the capability of showing that value? Especially in this case of wrong type values, know the content of `$value` would make debugging the cause extremely helpful.

Comment (dereuromark) [reply src/Database/Type/BoolType.php]: Yeah, maybe a bit out of scope - or even overkill - to use some Debugger::exportVar().
Hopefully the stracktrace contains the caller arguments.

**Comment by markstory** [comment]
Nice improvement :clap:

**Pull request merged**